Wikipedia/UI-V5/WMFArticleListTransition.m

```diff
@@ -25,43 +23,22 @@ @interface WMFArticleListTransition ()<UIGestureRecognizerDelegate>
 
 @implementation WMFArticleListTransition
 
-- (instancetype)initWithArticleListViewController:(WMFArticleListCollectionViewController*)listViewController articleContainerViewController:(WMFArticleContainerViewController*)articleContainerViewController contentScrollView:(UIScrollView*)scrollView {
+- (instancetype)initWithListCollectionViewController:(WMFArticleListCollectionViewController*)listViewController {
     self = [super init];
     if (self) {
-        _nonInteractiveDuration         = 0.5;
-        _dismissInteractively           = YES;
-        _listViewController             = listViewController;
-        _articleContainerViewController = articleContainerViewController;
-        _scrollView                     = scrollView;
-        [self addDismissGestureRecognizer];
+        self.listViewController = listViewController;
+        _nonInteractiveDuration = 0.5;
+        _isDismissing           = NO;
     }
     return self;
 }
 
-- (void)setDismissInteractively:(BOOL)dismissInteractively {
-    _dismissInteractively = dismissInteractively;
-    [self addDismissGestureRecognizer];
-}
-
-#pragma mark - UIViewControllerTransitioningDelegate
-
-- (id<UIViewControllerAnimatedTransitioning>)animationControllerForPresentedController:(UIViewController*)presented presentingController:(UIViewController*)presenting sourceController:(UIViewController*)source {
-    return self;
-}
-
-- (id<UIViewControllerAnimatedTransitioning>)animationControllerForDismissedController:(UIViewController*)dismissed {
-    return self;
-}
-
-- (id<UIViewControllerInteractiveTransitioning>)interactionControllerForPresentation:(id<UIViewControllerAnimatedTransitioning>)animator {
-    return nil;
+- (BOOL)isPresenting {
```

Comment:
@bgerstle while merging master into my pull - i looked at this again and remembered why this was 2 different BOOLs and not just the opposite of each other.

So isPresenting was supposed to only YES during the transition - when it is done presenting, isPresenting is NO since the presentation animation is finished. After presenting, isPresented is YES. At this time isDismissing is also NO since we are not dismissing either. So both isDismissing and isPresenting can both be NO at the same time.

I believe the isPresented variable was doing what you are trying to accomplish here now - but this was a separate concern. to put it another way. isPresented is about current state, isPresenting/isDismissing is about changing states. So to lay it out:

isPresented: Are we currently presented (regardless if a transition is running)?
isPresenting: Are we running the presentation transition animation?
isDismissing: Are we running the dismiss transition animation?